packages/tangle/messagestore.go

```diff
@@ -155,6 +161,25 @@ func (m *MessageStore) MissingMessages() (ids []MessageID) {
 	return
 }
 
+// StoreAttachment stores a new attachment if not already stored.
+func (m *MessageStore) StoreAttachment(transactionID ledgerstate.TransactionID, messageID MessageID) (cachedAttachment *CachedAttachment, stored bool) {
+	attachment, stored := m.attachmentStorage.StoreIfAbsent(NewAttachment(transactionID, messageID))
+	if !stored {
+		return
+	}
+	cachedAttachment = &CachedAttachment{CachedObject: attachment}
+	return
+}
+
+// Attachments retrieves the attachment of a transaction in attachmentStorage.
+func (m *MessageStore) Attachments(transactionID ledgerstate.TransactionID) (cachedAttachments CachedAttachments) {
```

Comment:
Since the attachments are immutable anyway we could always simply hand out a list of `Attachments` (type would need to be introduced) instead of cached objects. Makes it a bit easier to handle.

Reply:
I think It should follow the general pattern of the` messagestore` API. Methods return `cachedObjects`.

Reply:
Fair enough. We talked about simplifying it yesterday in a call with Hans and Angelo. Maybe this is something we can do later as a part of a bigger refactor.